### PR TITLE
CaseGroupingReport aggregation

### PR DESCRIPTION
**Pull request by kaapstorm**
## Technical Summary

This change adds filters and aggregation to `CaseGroupingReport` for geolocated cases.

Jira: [SC-3052](https://dimagi-dev.atlassian.net/browse/SC-3052)

## Feature Flag

GEOSPATIAL

### Safety story

Tested locally

### Automated test coverage

Includes tests

### QA Plan

Geospatial features will go through QA.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

[SC-3052]: https://dimagi-dev.atlassian.net/browse/SC-3052?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by kaapstorm** [comment]
Documentation around this work will be added to `corehq/apps/geospatial/README.md` in a follow-up PR.

**Comment by kaapstorm** [comment]
That failing test is not a mystery:
```
  ======================================================================
  FAIL: corehq.apps.geospatial.tests.test_es:TestGetMaxDocCount.test_max_doc_count
  ----------------------------------------------------------------------
  Traceback (most recent call last):
    File "/usr/local/lib/python3.9/unittest/mock.py", line 1336, in patched
      return func(*newargs, **newkeywargs)
    File "/mnt/commcare-hq-ro/corehq/apps/geospatial/tests/test_es.py", line 76, in test_max_doc_count
      self.assertEqual(max_doc_count, 3)
  AssertionError: 0 != 3
  -------------------- >> begin captured logging << --------------------
```
It is because my test decorator [`@flag_enabled('USH_CASE_CLAIM_UPDATES')`](https://github.com/dimagi/commcare-hq/blob/13a408dffd907726e40bc173d40e36e672aed082/corehq/apps/geospatial/tests/test_es.py#L38) is being ignored. If I remove [the feature flag check here](https://github.com/dimagi/commcare-hq/blob/f58767e90a0ff83b1ca401f3bcc9a399bdf687ec/corehq/pillows/case_search.py#L86), the test passes. :roll_eyes: Any idea why the test decorator is not doing what I expect?

File: corehq/apps/es/filters.py [hunk 1]
```diff
@@ -154,11 +154,74 @@ def geo_bounding_box(field, top_left, bottom_right):
...
+def geo_shape(
+    field: str,
+    shape: ElasticsearchGeoJSONGeometry,
+    relation: Literal[
+        'intersects',
+        'disjoint',
+        'within',
+        'contains',
+    ] = 'intersects',
+) -> dict[str, Any]:
```
Comment (zandre-eng) [inline corehq/apps/es/filters.py:207]: Are the type hints going to stay?

Comment (kaapstorm) [reply corehq/apps/es/filters.py:207]: Oops, I nearly forgot about those. No, I'll remove them.

**Comment by kaapstorm** [comment]
> It therefore seems like `case_search_es_setup()` is being called before `flag_enabled()`.

Interesting! Thank you! Using `flag_enabled()` as a context manager inside `setUpClass()` fixed it.

**Review by zandre-eng — APPROVED** [review]
Great job on getting through this tough PR!

**Pull request merged**